Core/include/Acts/Geometry/PortalShell.hpp

```diff
@@ -0,0 +1,225 @@
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Geometry/BoundarySurfaceFace.hpp"
+#include "Acts/Utilities/BinningType.hpp"
+#include "Acts/Utilities/Logger.hpp"
+
+#include <array>
+#include <memory>
+
+namespace Acts {
+
+class Portal;
+class TrackingVolume;
+class GeometryContext;
+
+/// @class PortalShellBase
+/// The portal shell of a volume is the set of portals that describes
+/// connections "outside" of the volume. Which boundary surfaces of a volume are
+/// included in the shell depends on the volume. Portal shells are only used
+/// during geometry construction, and are not part of the final geometry
+/// description.
+///
+/// This class is the base class for all portal shells
+class PortalShellBase {
```

Comment:
I was thinking about the name a bit if we could imply somehow that it is only used during construction but I could not come up with something better. Maybe `VolumePortalCollectorBase`?

Reply:
I'm not sure about the name either. I think shell implies that it carries geometric information as well, in the sense that the instance decides which faces are valid and where to assign them. So it's doing a bit more than collecting the portals.